janus_server/src/aggregator.rs

```diff
@@ -117,6 +125,7 @@ impl<C: Clock> Aggregator<C> {
             );
             return Err(Error::UnrecognizedMessage(
                 "unexpected number of encrypted shares in report",
+                report.task_id,
```

Comment:
Generally I agree on using `tracing_error`, but in this case the task ID is needed in the JSON document. I think it's more foolproof to pass it as part of the error enum variant, because we can't forget the ID when constructing an error. I would rather not fetch from a tracing span to assemble the response.